io-ballerina/src/io/bytes/readable_byte_channel.bal

```diff
@@ -17,13 +17,17 @@
 import ballerina/java;
 
 # ReadableByteChannel represents an input resource (i.e file). which could be used to source bytes.
+# A file path or an in-memory byte array can be used to obtain a ReadableByteChannel.
+# A ReadableByteChannel do not support initilization, and it should be obtained using the following methods or implement natively.
+# `io:openReadableFile("./files/sample.txt")` - used to obtain a ReadableByteChannel from a given file path
+# `io:createReadableChannel(byteArray)` - used to obtain a ReadableByteChannel from a given byte array
 public class ReadableByteChannel {
 
-    # Adding default init function to prevent object getting initialized from the user code.
+    # Adding default init function to prevent object getting initialized from the user code
     function init() {
     }
 
-    # Source bytes from a given input/output resource. The number of bytes returned will be < 0 if the file reached its end.
+    # Source bytes from a given input/output resource.
```

Comment:
An output resource?

Reply:
If I don't know how many bytes I need to read in advance, how do I tell the API to read until it reaches the end? Do I guess a size for `nBytes` and loop through till I come across EOF?

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-io/pull/37/commits/123095180ab4fe7865b5e69d8623895fe6b978e8